src/microbe_stage/Microbe.Contact.cs

```diff
@@ -505,14 +505,19 @@ public void Kill()
             }
         }
 
-        int chunksToSpawn = Math.Max(1, HexCount / Constants.CORPSE_CHUNK_DIVISOR);
+        // Queues either 1 corpse chunk or a factor of the organelles
+        int chunksToSpawn = Math.Max(1, organelles.Count / Constants.CORPSE_CHUNK_DIVISOR);
 
         var chunkScene = SpawnHelpers.LoadChunkScene();
 
+        // Duplicates the organelle list in a random order and creates an enumerator from it
+        var organellesAvailable = organelles.ToList().OrderBy(_ => random.Next());
```

Comment:
```suggestion
        var organellesAvailable = organelles.OrderBy(_ => random.Next()).GetEnumerator();
```
As the list is no longer updated, wouldn't this work? I mean not needing to make an intermediate list is going to be a benefit. You might need to slightly adjust how the enumerator call is done to compile.